src/execution/window_executor.cpp

```diff
@@ -878,18 +886,24 @@ WindowAggregateExecutor::WindowAggregateExecutor(BoundWindowExpression &wexpr, C
                                                  const ValidityMask &order_mask, WindowAggregationMode mode)
     : WindowExecutor(wexpr, context, count, partition_mask, order_mask), mode(mode), filter_executor(context) {
 
-	//	Check for constant aggregate
-	if (IsConstantAggregate()) {
-		aggregator = make_uniq<WindowConstantAggregator>(AggregateObject(wexpr), wexpr.return_type, partition_mask,
-		                                                 wexpr.exclude_clause, count);
-	} else if (IsCustomAggregate()) {
+	// Force naive for SEPARATE mode or for (currently!) unsupported functionality
+	AggregateObject aggr(wexpr);
+	if (mode == WindowAggregationMode::SEPARATE ||
+	    (wexpr.distinct && wexpr.exclude_clause != WindowExcludeMode::NO_OTHER)) {
+		aggregator = make_uniq<WindowNaiveAggregator>(aggr, wexpr.return_type, wexpr.exclude_clause, count);
```

Comment:
Perhaps we want to trigger this when optimization is disabled as well so that `PRAGMA enable_verification` automatically runs both implementations and compares the results?